### ISSUE TITLE
trace: Review & improve query for attribute values.

### ISSUE THREAD

**Issue by alanconway** [korrel8r/korrel8r#185]
Currently the trace store uses prometheus `/search` endpoint which does not return all attributes. The store adds a set of "select()" statements for a default set of attributes.
This may not include all attributes needed for correlations.

Possible solutions:
- Configure, or automatically discover (better), the set of all attribute names used by rules with the trace:span start class and use that in the store.
- Have the store use a combination of requests to `/search` (for trace IDs) and `/trace` (for complete trace data including all attributes). Careful to de-duplicate and avoid un-necessary requests.

**Comment by alanconway** [comment]
Currently the trace store uses prometheus `/search` endpoint which does not return all attributes. The store adds a set of "select()" statements for a default set of attributes.
This may not include all attributes needed for correlations.

Possible solutions:
- Configure, or automatically discover (better), the set of all attribute names used by rules with the trace:span start class and use that in the store.
- Have the store use a combination of requests to `/search` (for trace IDs) and `/trace` (for complete trace data including all attributes). Careful to de-duplicate and avoid un-necessary requests.

**Comment by too-common-name** [comment]
@alanconway 

**Related observation:** in addition to the missing attributes problem, the `/search` endpoint also limits the number of spans returned per spanset via the `spss` parameter (default: 3). This means that when correlating a specific trace by ID (e.g. `trace:span:{trace:id="..."}`) through `/search`, only up to 3 spans of the matching trace are returned, even if the full trace contains many more.

For example, a distributed trace spanning 3 services with 9 spans returns only 3 spans through `/search`, causing `TraceToPod` to discover only 1 pod instead of 3.

Looking at the current code in `tempo.go`, the `get()` method builds a query against the search endpoint without passing `spss`.

This strengthens the case for the second proposed solution: using `GET /api/traces/{traceID}` for queries that target a specific trace ID. The `/traces/{id}` endpoint returns the complete trace with all spans and all attributes, solving both the missing attributes and the `spss` truncation problems at once.

The `/search` endpoint remains appropriate for broader queries (e.g. filtering by `resource.k8s.pod.name`) where the results are expected to be a set of matching traces rather than a single complete trace.

**Comment by alanconway** [comment]
# Proposal
- new pseudo-class **trace:trace**, direct retrieval from /trace/{id} via korrel8r query **trace:trace:<id>**
- trace:trace is never represented as a korrel8r start object, the result of query `trace:trace:id` is a set of `trace:span` objects
- trace:span:<traceql> works as it does now via /search/{traceql}

Thoughts? Is there any value in having a representation of trace:trace other than as a set of spans? If we did represent them separately there'd be some rule duplication needed between starting from trace:trace and starting from trace:span

@too-common-name

**Comment by too-common-name** [comment]
@alanconway I agree — no value in representing trace:trace as a separate entity in the graph. The result should be a set of trace:span objects, so downstream rules (TraceToPod, TraceToLog, etc.) work unchanged without duplication.

The doc.go already documents two selector forms for trace:span — TraceQL ({...}) and trace IDs (bare hex). The store could branch on the selector format with a simple heuristic :

- Contains { → TraceQL → GET /api/search
- No { → trace IDs → GET /api/traces/{id}

This keeps the model simpler (one class, two query formats) but puts routing logic in the store. Your trace:trace approach is more explicit and declarative — the class name tells the store what to do, no heuristics (and honestly I prefer it). The tradeoff is that rule authors for LogToTrace-style rules need to know about the distinction, though in practice these rules live in quickrules so it is manageable.

**Comment by alanconway** [comment]
Good point about the alternate formats. I think I prefer that - trace:trace creates confusion as to "where are the trace:trace things"? and makes a special case pseudo-class which is icky. trace:span:ID is pretty clear - all the spans with this trace-ID.
